lib/streamlit/runtime/app_session.py

```diff
@@ -628,7 +628,9 @@ def _handle_scriptrunner_event_on_event_loop(
                 ), "exception must be set for the SCRIPT_STOPPED_WITH_COMPILE_ERROR event"
                 msg = ForwardMsg()
                 exception_utils.marshall(
-                    msg.session_event.script_compilation_exception, exception
+                    msg.session_event.script_compilation_exception,
+                    exception,
+                    is_uncaught_app_exception=False,
```

Comment:
In places calling `marshall` directly, I have added `is_uncaught_app_exception=False`. I think this should be the previous behaviour because we were only creating `UncaughtAppException` in `error_util.handle_uncaught_app_exception` so these areas calling `marshall` and enqueuing the message would not have entered that code path previously.

Reply:
It might be a little bit nicer to set the default value in the args for this parameter to `False` and only update the one relevant place where it diverges from the default: 

```python
def marshall(
    exception_proto: ExceptionProto,
    exception: BaseException,
    *,
    is_uncaught_app_exception: bool = False,
)
```

Reply:
Yeah, I was thinking of setting it explicitly everywhere, but this probably is safer.